### PR TITLE
Lu 6627 marshmallow

### PR DESCRIPTION
**Pull request by glanvl**
Added Marshmallow changes and updated the tests.
Also added some formatting (capital letter and full stops)
And single quotes changed to double.

### CODE REVIEW

File: tests/fixtures/test_wrangler_to_method_runtime.json [hunk 1]
```diff
@@ -1,25 +1 @@
...
-    "0076": "076"
-  },
-  "statuses": {
-    "Form Sent Out": 1,
-    "Clear": 2,
-    "Overridden": 2
-  }
-}
\ No newline at end of file
+{"data": null, "period": "201809", "periodicity": "03", "run_id": "bob", "question_labels": {"0601": "Q601_asphalting_sand", "0602": "Q602_building_soft_sand", "0603": "Q603_concreting_sand", "0604": "Q604_bituminous_gravel", "0605": "Q605_concreting_gravel", "0606": "Q606_other_gravel", "0607": "Q607_constructional_fill", "0608": "Q608_total"}, "survey_codes": {"0066": "066", "0076": "076"}, "statuses": {"Form Sent Out": 1, "Clear": 2, "Overridden": 2}}
```
Comment (KrisRogos) [inline tests/fixtures/test_wrangler_to_method_runtime.json:1]: perhaps you could run prettify on this file?

File: ingest_takeon_data_wrangler.py [hunk 2]
```diff
@@ -4,21 +4,60 @@
...
+
+    class Meta:
+        unknown = EXCLUDE
+
+    def handle_error(self, e, data, **kwargs):
+        logging.error(f"Error validating runtime params: {e}")
+        raise ValueError(f"Error validating runtime params: {e}")
+
+    in_file_name = fields.Str(required=True)
+    ingestion_parameters = fields.Nested(IngestionParamsSchema)
```
Comment (Piwington) [inline ingest_takeon_data_wrangler.py:51]: I believe you need required=True on this line as well?

File: ingest_takeon_data_wrangler.py [hunk 3]
```diff
@@ -4,21 +4,60 @@
...
+    survey_codes = fields.Dict(required=True)
+    statuses = fields.Dict(required=True)
+
+
+class RuntimeSchema(Schema):
+    """
+    Schema to ensure that runtime variables are present and in the correct format.
+    These variables are expected by the method, and it will fail to run if not provided.
+    :return: None
+    """
```
Comment (Piwington) [inline ingest_takeon_data_wrangler.py:41]: Happy with the docstring but thought I'd mention that the other modules don't bother with them for the Marshmallow Schema. I'll leave it to your judgement wether or not to leave it in.

File: ingest_takeon_data_wrangler.py [hunk 4]
```diff
@@ -27,49 +66,49 @@ def lambda_handler(event, context):
...
+        logger.info("Validated parameters.")
+
+        # Environment Variables.
+        checkpoint = environment_variables["checkpoint"]
+        takeon_bucket_name = environment_variables["takeon_bucket_name"]
+        method_name = environment_variables["method_name"]
+        results_bucket_name = environment_variables["results_bucket_name"]
+
+        # Runtime Variables.
+        runtime_variables = RuntimeSchema().load(event["RuntimeVariables"])
```
Comment (Piwington) [inline ingest_takeon_data_wrangler.py:93]: Is there as reason this isn't directly below the Environment Schema load like in the other modules?

File: tests/fixtures/test_wrangler_to_method_runtime.json (hunk 1 above)
Comment (Piwington) [reply tests/fixtures/test_wrangler_to_method_runtime.json:1]: Yeah could you pretty please :)

File: ingest_takeon_data_wrangler.py (hunk 3 above)
Comment (glanvl) [reply ingest_takeon_data_wrangler.py:41]: I'll remove it for some reason this module already had it there so I left it in.

File: ingest_takeon_data_wrangler.py (hunk 4 above)
Comment (glanvl) [reply ingest_takeon_data_wrangler.py:93]: No not sure why think I was just keeping runtime with runtime and env with env.

**Pull request merged**